README.md

```diff
@@ -1,11 +1,202 @@
-# lambda-template
-A GitHub template for quickly starting a new AWS lambda project.
-
-## Naming
-Naming conventions:
-* for a vanilla Lambda: `lambda-<context>`
-* for a Cloudformation Transform macro: `cfn-macro-<context>`
-* for a Cloudformation Custom Resource: `cfn-cr-<context>`
+# lambda-finops-cost-rules
+A lambda for generating the JSON string used by the Rules property of an
+AWS::CE::CostCategory CloudFormation resource.
+
+## Design
+This lambda is intended to transform the chart of accounts returned from
+[lambda-mips-api](https://github.com/Sage-Bionetworks-IT/lambda-mips-api),
+combined with current AWS account tags, into cost-category rules (JSON string)
+appropriate to pass to the [`Rules` property for `AWS::CE::CostCategory`](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ce-costcategory.html#cfn-ce-costcategory-rules).
+
+This lambda will query `lambda-mips-api` for a chart of accounts (a mapping of
+program codes and program names), and query AWS Organizations for account-level
+cost-tracking tags.
+
+Cost categories are created for each program listed in the chart of accounts.
+
+Tags may exist on either individual cloud resources or on their containing
+accounts which assign the resource costs to a particular program category.
+A list of tag names that may contain program assignments are provided as a
+template parameter.
+
+### Category Rules
+
+Cost-category rules are created to assign line item costs to a category based on
+tags for the line item resource or its containing account.
+
+First, this lambda will create cost-category rules that check a list of tags
+(provided as a template parameter) for every combination of program code and
+tag name, also create rules for each account with an appropriate tag, and
+finally create rules to inherit the value from a listed tag if no other rule
+matches.
+
+The order of the rules matters because processing will stop at the first match.
+For more information on rule processing, see the [AWS User Guide](https://docs.aws.amazon.com/awsaccountbilling/latest/aboutv2/manage-cost-categories.html).
+
+### Components
+![Component Diagram](docs/component.diagram.png)
```

Comment:
are you going to setup cloudfront for this lambda?  the diagram shows it but it doesn't seem necessary for this lambda because it's not making a direct request to mips, correct?

Reply:
yeah, I suppose its a little overkill, I don't need to copy that part from the mips lambda